### ISSUE TITLE
Allow aria-required on fieldset?

### ISSUE THREAD

**Issue by WilcoFiers** [dequelabs/axe-core#2558]
This question came in through twitter:

> https://twitter.com/clcorde/status/1316048183299112967
> 
> Using aria-required attr on `<fieldset>`
> `<fieldset aria-required="true">` AXE is firing an error: aria attribute not allowed. I looked at the technical documentation and it seems to be supported for radiogroup role; is this contradictory?

The answer is that `fieldset` maps to the `group` role, which does not have aria-required. When I tested this out though, I couldn't get aria-required to work on fieldsets at all, not even by giving it role="radiogroup". I only tested with Safari+VO. Maybe I'm missing something, but it looks like there's a problem with the way axe-core does aria-required. We'll need to do some research. Here's the code I tried:

```html
<fielset role="radiogroup" aria-required="true">
	<legend>Favorite H. Jon Benjamin show</legend>
	<input type="radio" id="bob" /><label for="bob">Bob's burgers</label>
	<input type="radio" id="archer" /><label for="archer">Archer</label>
</fielset>
```

**Comment by straker** [comment]
Tested this and only JAWS/IE and JAWS/Chrome report that the radio buttons are required. NVDA/Firefox, VO/Safari, and Android Voice Assistant/Chrome do not report required.

**Comment by straker** [comment]
We can add an `unsupportedAttrs` property to the `aria-roles.js` file that flags certain combination of roles/attributes as unsupported due to lack of browser support. Make it's own check and add to the `aria-allowed-attr` rule.

**Comment by straker** [comment]
Closing as this is just a single use case with a lot of special logic behind it

**Comment by riteshjagga** [comment]
We need this too, reason being fieldset gives an ability to disable all the radio buttons using the disabled attribute.

For a group of radio buttons the **required** word should be announced one by the screen reader which is possible only if we allow `aria-required` on a `fieldset`. Adding `aria-required` on every radio button will announce it every time radio button is focused/selected which could cause confusion.

So, I think both `role="radiogroup"` and `aria-required` should be allowed.

**Comment by straker** [comment]
@riteshjagga you should be able to add `role="radiogroup"` to the fieldset to be able to add the `aria-required` attribute. 

```html
<fieldset role="radiogroup" aria-required="true">
  <!-- Radio button 1 -->
  <!-- Radio button 2 -->
  <!-- Radio button 3 -->
</fieldset>
```

**Comment by riteshjagga** [comment]
Thank you @straker. I think TypeScript and spread operator made it difficult to identify the issue clearly. 

I had the below code where issue was reported.

```
  const a11yProps = {
    role: role,
    ...(required && {'aria-required': 'true'}),
  }

<fieldset {...a11yProps}>
  <legend>Some legend</legend>
</fieldset>
```

I changed string `'true'` to boolean `true` as in `...(required && {'aria-required': true}),` to fix it.
